README.md

```diff
@@ -36,3 +36,158 @@ This is the shared library defining public classes that constitute the worker in
+
+The event obj `TaskEventObject` contains below parameters. The variables will be initialized with values from task message and it will be sent to the series of other functions in the workflow for further processing.
+```
+TaskEventObject(Task task){
+    application=task.application;
+    task = task;
+    rootDocument=task.document;
+}
+```
+For more details of the TaskEventObject refer the java implementation of the class from package  `com.hpe.caf.worker.document.scripting.events.TaskEventObject`.
+
+```
+/* global thisScript */
+
+function onProcessTask(e)
+{	
+    // e.application  (read-only)
+    // e.task         (read-only)
+    // e.rootDocument (read-only)
+}
+```
+
+#### onBeforeProcessDocument
+
+This event will be executed after `onProcessTask` and before processing of a document. 
+
+This function is passed `CancelableDocumentEventObject`  with `Document` as an argument.
+
+The CancelableDocumentEventObject has below variables.
+```
+CancelableDocumentEventObject(Document document)
+{
+    super(document);
+    cancel=false;
+}
+```
+For more details of the event object refer the Java implementationof the class from package `com.hpe.caf.worker.document.scripting.events.CancelableDocumentEventObject`.
+```
+function onBeforeProcessDocument(e)
+{
+    // e.application  (read-only)
+    // e.task         (read-only)
+    // e.rootDocument (read-only)
+    // e.document     (read-only)
+    // e.cancel       (writable)  (default: false)
+}
+```
+Output of this function would be the boolean value of event's cancel flag.
+This flag is used to determine if that individual document should be processed by the worker.
+
+#### onProcessDocument
+
+This function will be called to process a document. 
+This function is passed `DocumentEventObject` with `Document` as an argument.
+The `onBeforeProcessDocument` function already triggered to check for cancellation flag.
+
+The `DocumentEventObject` has below variables.
+```
+DocumentEventObject(Document document){
+    document = document;
+}
+```
+For more details  of the event object refer the java implementation from the class `com.hpe.caf.worker.document.scripting.events.DocumentEventObject`.
+```
+function onProcessDocument(e)
+{
+    // e.application  (read-only)
+    // e.task         (read-only)
+    // e.rootDocument (read-only)
+    // e.document     (read-only)
+}
+```
+This function calls the customization scripts, and if none of them have set the cancellation flag then calls the implementation's `DocumentWorker#processDocument processDocument()}` function. The response of this function will be the processed document that should be add to the batch of documents.
+
+#### onAfterProcessDocument
+
+This function will be called once the processing of the document completed successfully.
+This function is passed `DocumentEventObject` with `Document` as an argument.
+The `DocumentEventObject` has below variables.
+
+```
+DocumentEventObject(Document document){
```

Comment:
Same as above, also we probably don't need two examples of the same object in the one section of the documentation.